lib/NetPacksLib.cpp

```diff
@@ -594,7 +594,11 @@ DLL_LINKAGE void HeroRecruited::applyGs( CGameState *gs )
 
 	h->setOwner(player);
 	h->pos = tile;
-	h->movement =  h->maxMovePoints(true);
+	bool fresh = (static_cast<int>(h->movement) == -1);

```

Comment:
If you use same -1 value in very different files, I would replace it with constant or ENUM for clarity.